parsec/crypto.py

```diff
@@ -45,40 +45,62 @@
 # Types
 
 
-class SecretKey(bytes):
-    __slots__ = ()
+class SecretKey:
+    __slots__ = ("_secret",)
+
+    def __init__(self, secret: bytes):
+        self._secret = secret
 
     def __new__(cls, rawkey: bytes) -> "SecretKey":
         if len(rawkey) != SecretBox.KEY_SIZE:
             raise ValueError("Invalid key size")
-        return super(SecretKey, cls).__new__(cls, rawkey)
+        return super().__new__(cls)
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self.secret == other.secret
 
     @classmethod
     def generate(cls) -> "SecretKey":
         return cls(random(SecretBox.KEY_SIZE))
 
     def __repr__(self):
         # Avoid leaking the key in logs
-        return f"<{type(self).__module__}.{type(self).__qualname__} object at {hex(id(self))}>"
+        return "SecretKey"
```

Comment:
Why changing the existing implementation?

Reply:
we cannot use the existing implementation until secretKey and SigningKey are finished because of `tests/test_types.py::test_keys_dont_leak_on_repr`

Reply:
so is there a TODO to add ? from my point of view it is fine to keep the repr of all the crypto classes to something dead simple like you did.

I would just display it as something like `SecretKey(<redacted>)` (given the parenthesis with attributes inside is the most common way to represent an object)